src/client/metric/mod.rs

```diff
@@ -875,8 +869,9 @@ impl<T: MetricType> MMVWriter for InstanceMetric<T> {
         let instance_blk_offs = write_indom_and_instances(ws, c, &self.indom, mmv_ver)?;
 
         // write value blocks
-        for ((_, instance), instance_blk_off) in self.vals.iter_mut().zip(instance_blk_offs) {
```

Comment:
This was a bug because the iteration order of `self.vals`, which is a `HashSet`, was undefined, and so value blocks were being mapped to the wrong instance blocks.

Reply:
any particular reason on sticking with `HashSet` and not going for `TreeSet` for a deterministic iteration order?

Reply:
It would be deterministic, but it wouldn't align with a Vec (populated by iterating a different set) anyway. 

The only benefit I see for deterministic iteration is that the order of instances in an mmvdump would be deterministic as well. I can do that in another PR.

Reply:
That's not needed IMO